libcnb/src/layer_env.rs

```diff
@@ -262,7 +262,8 @@ impl LayerEnv {
                 let target_delta = match target_lifecycle {
                     TargetLifecycle::Build => &mut result_layer_env.layer_paths_build,
                     TargetLifecycle::Launch => &mut result_layer_env.layer_paths_launch,
-                    _ => panic!("Unexpected TargetLifecycle in read_from_layer_dir implementation. This is a libcnb implementation error!"),
+                    // Other `TargetLifecycle` variants are not valid in `layer_path_specs` and would indicate a libcnb bug.
+                    _ => unreachable!(),
```

Comment:
Do we want to keep the previous message? It would ease debugging should that match arm ever match.

```suggestion
                    _ => unreachable!("Unexpected TargetLifecycle in read_from_layer_dir implementation. This is a libcnb implementation error!"),
```

Reply:
@Malax Ah! I didn't know a custom message was possible with `unreachable!` since the docs don't mention it (and I wasn't yet familiar with the nuances of `macro_rules` syntax; I've now read up on that):
https://doc.rust-lang.org/core/macro.unreachable.html

I've opened an upstream PR to improve the docs:
https://github.com/rust-lang/rust/pull/90798